chargebyte_board.py

```diff
@@ -0,0 +1,280 @@
+
+
+class StatusPWMGeneration(Enum):
+    PWM_GENERATION_IS_DISABLED = 0
+    PWM_GENERATION_IS_ENABLED = 1
+
+
+class SocketAction(Enum):
+    UNLOCK_SOCKET = 0
+    LOCK_SOCKET = 1
+    QUERY_LIMIT_SWITCH_STATUS = 2
+    RESERVED = 3  # Reserved values from 3 to 255
+
+
+class LockStatus(Enum):
+    OPEN = 0
+    CLOSED = 1
+    NOT_CONNECTED = 2
+
+
+class ErrorCode(Enum):
+    NO_ERROR = 0
+    INVALID_PARAMETER = 1
+
+
+class StatusCode(Enum):
+    OFF = 0
+    ACTIVE = 1
+
+
+class ChargebyteException(Exception):
+    def __init__(self, message):
+        self.message=message
+
+
+class ChargebyteBoard:
+
+    def __init__(self, host: str, port: int):
+        self.s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.s.connect((host, port))
+        self.s.settimeout(15.0)
+        self.s.setblocking(0)
+        self.s.lock = Lock()
+
+
+    def send_packet(self, service_id: int, payload: bytearray)->bytearray:
+        self.s.send(self.build_message(service_id, payload))
+        response = self.read_response()
+        self.check_response(service_id, response)
+        return self.parse_response(response)
+
+
+    def build_message(self, service_id: int, payload: bytearray)->bytearray:
+        if not isinstance(service_id,int):
+            raise TypeError
+        start_of_message = 0x02
+        length_of_message = 0x03 + len(payload)
+        device_adress = 0x00
+        response = bytearray([start_of_message,length_of_message,device_adress,service_id])+payload
+        return response+bytearray([self.check_block_sum(response)])
+
+
+    def read_response(self)->bytearray:
+        beginning = self.s.recv(1)
+        length = self.s.recv(1)
+        data = self.s.recv(int(length))
+        return bytearray([beginning,length])+data
+
+
+    def check_response(self, service_id:int, response:bytearray)->None:
+        if response[0] != 0x02:
+            raise ChargebyteException('beginning of message was not 0x02')
+        if response[3] != service_id + 0x80:
+            raise ChargebyteException('this response does not corresponds to the service that was requested')
+        if self.check_block_sum(response[:-1]) != response[-1]:
+            raise ChargebyteException('Something went wrong: the check block is wrong!')
+
+
+    def check_response_length(self, response:list, length:int)->None:
+        if len(response) != length:
+            raise Exception('Something went wrong, the response has an unexpected length!')
+
+
+    def parse_response(self, response:bytearray)->bytearray:
+        return response[4:-1]
+
+
+    def check_block_sum(self, numbers:bytearray)->int:
+        response = numbers[0]
+        for num in numbers[1:]:
+            response = response ^ num
+        return response
+
+
+    def join_bytes(self, low_byte:int, high_byte:int)->int:
+        return (high_byte << 8) | low_byte
+
+
+    def keep_service_alive():
+        pass
+
+
+    def test_device_one(self)->int|int|Enum:
```

Comment:
Using `|` means you return one of `int`, `int`, or `Enum`, but in reality you return all three of them as a tuple.
Also make sure to annotate the actual enum type not the generic `Enum`.

in this case use `tuple[int, int, ResetType]` as the type annotation

Reply:
thank you, i didnt know that, i was given the wrong information. will fix it.